### PR TITLE
[NEW] Support for dynamic slack and rocket.chat channels

### LINKED ISSUE #6509: Slackbridge one way if channel was created in Rocket first
Rocket.Chat Version: 0.49.3

Creating a channel in Rocket.Chat first and then in Slack leads to a one way synchronisation.
Chatbot joined in Slack by invite, message shows up in Rocket. All further messages are synced from Slack to Rocket but not from Rocket to Slack.

Creating it in Slack then in Rocket works seamlessly.

### PR DESCRIPTION
**Pull request by Hudell**
@RocketChat/core 

Closes an item on #8012 , merging PR #6282
Also closes #6509
Original description: 

> The goal of this pull request was to support Slack and Rocket channels coming and going. You had to enable/disable the slack bridge for it to pick up on changes.
> Refactored the code into multiple files to allow for easier contributions in the future.
> Fixed a 'not defined' error that was polluting the server logs
> Improved logging/debugging messages some more

### CODE REVIEW

**Comment by rodrigok** [comment]
@Hudell can you fix the conflict?

File: packages/rocketchat-slackbridge/package.js [hunk 1]
```diff
@@ -12,9 +12,14 @@ Package.onUse(function(api) {
...
 	api.addFiles('server/logger.js', 'server');
 	api.addFiles('server/settings.js', 'server');
 	api.addFiles('server/slackbridge.js', 'server');
 	api.addFiles('server/slackbridge_import.server.js', 'server');
+	api.addFiles('server/RocketAdapter.js', 'server');
+	api.addFiles('server/SlackAdapter.js', 'server');
+});
+
+Npm.depends({
+	'slack-client': '2.0.6'
```
Comment (engelgabriel) [inline packages/rocketchat-slackbridge/package.js]: Why are we adding this here?

File: packages/rocketchat-slackbridge/server/slackbridge.js [hunk 2]
```diff
@@ -1,1365 +1,99 @@
...
-					};
-
-					RocketChat.models.Messages.setPinnedByIdAndUserId(`slack-${ pin.channel }-${ pin.message.ts.replace(/\./g, '-') }`, msgObj.u, true, new Date(parseInt(pin.message.ts.split('.')[0]) * 1000));
-				}
-			}
-		}
-	}
+		//Choose whether SlackBridge should also send your messages back to Slack
+		RocketChat.settings.get('SlackBridge_Out_Enabled', (key, value) => {
+			logger.class.debug(`Setting: ${ key }`, value);
```
Comment (rodrigok) [inline packages/rocketchat-slackbridge/server/slackbridge.js]: Should this do something?

File: packages/rocketchat-slackbridge/server/slackbridge.js [hunk 3]
```diff
@@ -1,1365 +1,99 @@
...
 
-	importMessages(rid, callback) {
-		logger.class.info('importMessages: ', rid);
-		const rocketchat_room = RocketChat.models.Rooms.findOneById(rid);
-		if (rocketchat_room) {
-			if (this.slackChannelMap[rid]) {
-				this.copySlackChannelInfo(rid, this.slackChannelMap[rid]);
+		//Send messages from all channels that exist in Slack and the bot has joined
+		RocketChat.settings.get('SlackBridge_Out_All', (key, value) => {
+			logger.class.debug(`Setting: ${ key }`, value);
```
Comment (rodrigok) [inline packages/rocketchat-slackbridge/server/slackbridge.js]: Should this do something?

File: packages/rocketchat-slackbridge/server/slackbridge.js [hunk 4]
```diff
@@ -1,1365 +1,99 @@
...
+		});
 
-				logger.class.debug('Importing messages from Slack to Rocket.Chat', this.slackChannelMap[rid], rid);
-				let results = this.importFromHistory(this.slackChannelMap[rid].family, { channel: this.slackChannelMap[rid].id, oldest: 1 });
-				while (results && results.has_more) {
-					results = this.importFromHistory(this.slackChannelMap[rid].family, { channel: this.slackChannelMap[rid].id, oldest: results.ts });
-				}
+		//Choose which channels will send messages back to Slack
+		RocketChat.settings.get('SlackBridge_Out_Channels', (key, value) => {
+			logger.class.debug(`Setting: ${ key }`, value);
```
Comment (rodrigok) [inline packages/rocketchat-slackbridge/server/slackbridge.js]: Should this do something?

**Pull request merged**